### PR TITLE
Rename `admission_date` to `encounter_date` and clean column

### PR DESCRIPTION
**Pull request by rithviknishad**
## Proposed Changes

- Rename `admission_date` to `encounter_date`
- Updated validations to allow `encounter_date` for all consultations
- ⚠️ **Custom migration**: set `encounter_date` to `created_date` for existing consultations that are not admission/domiciliary care or `encounter_date` is `null`.
- Alter field: `encounter_date` is non-nullable and defaults to `now`
- Discharge Summary: show both date and time of admission

### Associated Issue

- https://github.com/coronasafe/care_fe/issues/6669
- https://github.com/coronasafe/care_fe/issues/6668
- Fixes https://github.com/coronasafe/care_fe/issues/6727
- FE: https://github.com/coronasafe/care_fe/pull/6719

## Merge Checklist

- [x] Tests added/fixed
- [ ] Update docs in `/docs`
- [x] Linting Complete
- [ ] Any other necessary step

_*Only PR's with test cases included and passing lint and test pipelines will be reviewed*_

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

File: care/facility/models/patient_consultation.py [hunk 1]
```diff
@@ -126,7 +127,7 @@ class PatientConsultation(PatientBaseModel, ConsultationRelatedPermissionMixin):
     referred_by_external = models.TextField(default="", null=True, blank=True)
     is_readmission = models.BooleanField(default=False)
     admitted = models.BooleanField(default=False)  # Deprecated
-    admission_date = models.DateTimeField(null=True, blank=True)  # Deprecated
+    encounter_date = models.DateTimeField(default=timezone.now)
```
Comment (sainak) [inline care/facility/models/patient_consultation.py:130]: We should index this field

**Review by sainak — CHANGES_REQUESTED** [review]
LGTM

**Review by sainak — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Comment by vigneshhari** [comment]
Resolve Conflicts.

**Review by vigneshhari — APPROVED** [review]

**Pull request merged**